src/program-editors.rst

```diff
@@ -22,6 +22,127 @@ Visual Programming Languages
+core to most visual programming languages indent to provide end-user programming.
+
+
+Motivations
+~~~~~~~~~~~
+
+In terms of the editing aspects of visual dataflow languages, usability improvement is what we most care about since that's where the programming efficiency increases. By applying Cognitive Dimensions framework to evaluate the visual 
+dataflow language tools like LabVIEW, it is obvious that visual dataflow programming language has a higher usability than tradional programming languages, espeically for end users with limited understanding of programming. Another point 
+that programmers / researchers feel excited about visual dataflow programming is that the execution sequence of a data flow program is not determined by a program counter taking through an ordered series of instructions as we would have 
+in Von Neumann style program but is instead contingent upon data dependencies becoming satisfied, which actually allows automatic parallelism to become realizable without traditional multi-threaded programming techniques.
+
+.. container:: bib-item
+
+  .. bibliography:: notation.bib
+    :filter: key == 'vpl1996usability'
+
+This article introduces the ‘Cognitive Dimensions’ Framework technique to evaluate the usability of Visual Programming Environments. The researched tools are LabVIEW and Prograph.
+
+.. container:: bib-item
+
+  .. bibliography:: notation.bib
+    :filter: key == 'dpl2014compiler'
+
+This article illustrate how a compiler framework can enable the implementation of a stream system called COStream which is based on the dataflow programming paradigm that improves programming productivity.
+
+Applications
+~~~~~~~~~~~~
+
+LabVIEW
+^^^^^^^
+
+LabVIEW (Laboratory Virtual Instrument Engineering Workbench) is a system-design platform and development environment for a visual programming language designed by National Instruments based on visual data-flow language paradigm.
+LabVIEW followed the convention of a box-line structure and enriched it with constructs for conditionals, loops and sequences. And different types of lines represent different types of data (numbers, booleans, strings, arrays ...) 
+flowing among boxes. The operators provided perfectly covered almost all needs to construct a virtual electronic instrument operating in real time but make considerably less coverage provision for other domains of computing.
+Besides the usability, LabVIEW has been keeping a great contribution on researches about cognitive effects of visual programming representations.
+
+.. container:: bib-item
+
+  .. bibliography:: notation.bib
+    :filter: key == 'vp2001labview'
+
+This article reports the results of an exploratory survey of experienced LabVIEW
+users, a main goal being to learn how LabVIEW programmers think that the visual
+representation of LabVIEW affects their programming.
+
+Prograph
+^^^^^^^^
+
+Prograph is a visual data flow and object-orientated language marketed by TGS Systems.
```

Comment:
Only include a system if there is something to cite and if it is a notable system. What is unique about Prograph?